Lib/test/test_using.py

```diff
@@ -107,3 +107,62 @@ def test_region_cown_ptr(self):
     def test_invalid_cown_init(self):
          # Create cown with invalid init value
         self.assertRaises(RegionError, Cown, [42])
+
+    def test_threads(self):
+        from threading import Thread
+        from using import using
+
+
+        class Counter(object):
+            def __init__(self, value):
+                self.value = value
+
+            def inc(self):
+                self.value += 1
+
+            def dec(self):
+                self.value -= 1
+
+            def __repr__(self):
+                return "Counter(" + str(self.value) + ")"
+
+
+        # Freezes the **class** -- not needed explicitly later
+        makeimmutable(Counter)
+
+        def ThreadSafeValue(value):
+            r = Region("counter region")
+            r.value = value
+            c = Cown(r)
+            # Dropping value, r and closing not needed explicitly later
+            del value
+            del r
+            c.get().close()
+            return c
+
+        def work(c):
+            for _ in range(0, 100):
+                c.inc()
+
+        def work_in_parallel(c):
+            @using(c)
+            def _():
+                work(c.get().value)
```

Comment:
When we talked, you explained that this test verifies the concurrent nature by not deadlocking. Or better say, it shows that it likely works, since there is a good chance that it would deadlock otherwise.

But shouldn't this loop contain the `@using(c)...` function to proof this?

Currently, it looks like the `work_in_parallel` acquires the cown once and then finishes all 100 incs. Wouldn't it be a better test to aquire and release the cown 100 times per thread?